app/models/manageiq/providers/azure_stack/cloud_manager/event_target_parser.rb

```diff
@@ -0,0 +1,43 @@
+class ManageIQ::Providers::AzureStack::CloudManager::EventTargetParser
+  include ManageIQ::Providers::AzureStack::EmsRefMixin
+
+  attr_reader :ems_event
+
+  # @param ems_event [EmsEvent] EmsEvent object
+  def initialize(ems_event)
+    @ems_event = ems_event
+  end
+
+  # Parses all targets that are present in the EmsEvent given in the initializer
+  #
+  # @return [Array] Array of InventoryRefresh::Target objects
+  def parse
+    parse_ems_event_targets(ems_event)
+  end
+
+  # Parses list of InventoryRefresh::Target out of the given EmsEvent
+  #
+  # @param event [EmsEvent] EmsEvent object
+  # @return [Array] Array of InventoryRefresh::Target objects
+  def parse_ems_event_targets(event)
+    target_collection = InventoryRefresh::TargetCollection.new(:manager => event.ext_management_system, :event => event)
+
+    parse_event_target(target_collection, event)
+    targets = target_collection.targets
+
+    msg = "Mapped [#{event[:event_type]}] to refresh targets #{targets.map(&:manager_ref)}"
+    $azure_stack_log.debug(msg)
+
+    targets
+  end
+
+  def parse_event_target(target_collection, event)
+    resource_id = event[:full_data][:resource_id]
+    unless resource_id.nil?
+      if event[:event_type].downcase.include?("succeeded")
+        group_ref = resource_group_id(resource_id)
+        target_collection.add_target(:association => :resource_groups, :manager_ref => {:ems_ref => group_ref})
```

Comment:
I don't see how it would make sense to wrap this piece of code into the IF statement. It's not up to parser to decide what type of events it wants to support - that decision is made on higher level (in Automate, to be specific). You should imagine the parser like this (a very big picture):

```
parser(EVENT) => RESOURCE_GROUP_EMS_REF
```

So for _any_ event, regardless its type, it should calculate the affected ResourceGroup and list it as refresh target. Then it's up to Automate to decide what event types it wants to invoke the parser for, and what not.

Reply:
I see, thanks for the warning and explanation, I was missing this bit. I updated this piece of code so that we're now adding a resource group target for any event, as long as we're able to determine the resource group from the event data.